test/readConfig.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const metatests = require('metatests');
+
+const { clone } = require('@metarhia/common');
+
+const {
+  /* eslint-disable-next-line no-unused-vars */
+  default: { decorators, ...defParam },
+  fs: { load, readConfig },
+} = require('../metaschema');
+
+const { getSchemaDir } = require('./utils');
+
+metatests.test(
+  'Must properly validate application schemas by system schemas',
+  async test => {
+    /* eslint-disable-next-line no-unused-vars */
+    let updatedParam, ms, modulesMs, incorrectMs;
+    const systemParam = clone(defParam);
+    const param = clone(defParam);
+
+    const path = getSchemaDir('schemaValidationSimple');
+    const aPath = getSchemaDir('applicationSchema');
+    const iPath = getSchemaDir('incorrectApplicationSchema');
+    const mPath = getSchemaDir('modules');
+
+    try {
+      updatedParam = await readConfig(path, systemParam, param);
+      ms = await load(aPath, updatedParam.options, updatedParam.config);
+      modulesMs = await load(mPath, updatedParam.options, updatedParam.config);
+    } catch (error) {
+      test.fail(error);
+      test.end();
+      return;
+    }
+
+    try {
+      incorrectMs = await load(
+        iPath,
+        updatedParam.options,
+        updatedParam.config
+      );
+      test.fail();
+    } catch (err) {
+      test.throws(err);
+    }
+
+    test.strictSame(
+      updatedParam.config.processors.domains.hasOwnProperty('validateInstance'),
+      true
+    );
+    test.strictSame(
+      updatedParam.config.processors.category.hasOwnProperty(
+        'validateInstance'
+      ),
+      true
+    );
+
+    test.strictSame(ms.domains.size, 2);
```

Comment:
This test is redundant, `test/load` already checks that schemas are loaded correctly.